meter/shelly/connection.go

```diff
@@ -3,27 +3,31 @@ package shelly
 import (
 	"errors"
 	"fmt"
-	"net/http"
 	"strings"
+	"time"
 
 	"github.com/evcc-io/evcc/util"
 	"github.com/evcc-io/evcc/util/request"
 	"github.com/evcc-io/evcc/util/transport"
-	"github.com/jpfielding/go-http-digest/pkg/digest"
 )
 
 // Connection is the Shelly connection
 type Connection struct {
 	*request.Helper
-	uri     string
-	channel int
-	gen     int    // Shelly api generation
-	model   string // Shelly device type
-	profile string // Shelly device profile
+	log                util.Logger
+	uri                string
+	channel            int
+	gen                int    // Shelly api generation
+	model              string // Shelly device type
+	profile            string // Shelly device profile
+	Cache              time.Duration
+	gen1Status         util.Cacheable[Gen1Status]
```

Comment:
Ab hier wirds unschön. Anstatt 3 disjunkte Caches zu haben wäre es schön, Gen1+2 ein gemeinsames Interface zu geben. Jede  Implementierung des Interfaces kann dann seinen eigenen    Cache haben und hier embedded werden:

```go
type Gen interface{ ... }

type Connection struct {
	*request.Helper
    ...
    Gen
}

var gen Gen
if conn.gen < 2 {
    gen := &Gen1{
        cache: util.NewCacheable[Gen1Status](),
    }
}

conn := &Connection{
    Helper:  client,
    log:     *log,
    uri:     util.DefaultScheme(uri, "http"),
    channel: channel,
    gen:     resp.Gen,
    model:   strings.Split(resp.Type+resp.Model, "-")[0],
    profile: resp.Profile,
    Gen: gen
}
```

Reply:
Und die Caches müssen initialisiert werden, sonst gibts ein panic.

Reply:
ok, versuch ich entsprechend umzusetzen, kann sein, dass du mir mit ein paar Tipps bei der Umsetzung helfen musst. 
Könntest du mir noch einmal eine Einladung für Slack schicken? Slack hat mich aufgrund von zu langer Inaktivität rausgeworfen ...

Reply:
Du meldest dich da einfach selbst wieder an, braucht keine Einladung

Reply:
@andig kannst du dir den aktuellen Stand des PR einmal anschauen? Die mir möglichen Tests sehen gut aus. Durch die Einführung der Interface Funktionen für die beiden Shelly API Generationen ist der Diff etwas größer geworden.

Offen ist noch die Korrektur evtl. negativer Power Messwerte in `meter/shelly.go` im Falle des Einsatzes eines Shellies als PV-Meter (usage=pv). Soll / kann ich das noch einbauen?

Reply:
Klar- wäre super wenn zus. Logik dabei draussen bliebe